src/string_object.cpp

```diff
@@ -671,6 +679,32 @@ Value StringObject::lstrip(Env *env) const {
     }
 }
 
+Value StringObject::lstrip_bang(Env *env) {
+    if (is_frozen()) {
+        auto frozenError = GlobalEnv::the()->Object()->const_find(env, SymbolObject::intern("FrozenError"))->as_class();
+        env->raise_exception(new ExceptionObject(frozenError));
+    }
+
+    if (length() == 0)
+        return NilObject::the();
+
+    assert(length() < NAT_INT_MAX);
+    nat_int_t first_char;
+    nat_int_t len = static_cast<nat_int_t>(length());
+    for (first_char = 0; first_char < len; first_char++) {
+        char c = c_str()[first_char];
+        if (!is_strippable_whitespace(c))
+            break;
+    }
+
+    if (first_char == 0)
+        return NilObject::the();
+
+    memcpy(&m_string[0], &m_string[0] + first_char, len - first_char);
```

Comment:
You cannot memcpy from one part of a string onto an overlapping part. From the man page for `memcpy`:

>If dst and src overlap, behavior is undefined.

So we'll need a temporary buffer.

Reply:
Oh yes, wierldy I have been sure that is defined for this kind of copy, but we can use [memmove](https://en.cppreference.com/w/cpp/string/byte/memmove) without temporary buffer

Reply:
OK, I think I see. It's not actually overlapping, since it's copying characters to the "left". Ok, carry on.